initial additions for URL parsing

georgesnow opened the pull request:
Can't fully test getting "Unsupported plugin error". I tried doing fresh pull and compile of MacPass, but no change. Plugin builds successfully. 

However, force loading incompatible plugins shows in the NSLog the right value is being captured and sent.
Example: https://subdom.root.com/this/page.html
Returned value: subdom.root.com

Left the NSLog info in and only applied to Safari. I didn't have Chrome installed to test.

georgesnow committed:
initial additions for URL parsing

Can't fully test getting "Unsupported plugin error". I tried doing fresh pull and compile of MacPass, but no change. Plugin builds successfully. 

However, force loading incompatible plugins shows in the NSLog the right value is being captured and sent.
Example: https://subdom.root.com/this/page.html
Returned value: subdom.root.com

Left the NSLog info in and only applied to Safari. I didn't have Chrome installed to test.

mstarke commented on AutotypeURL/ATUAutotypeURL.m:
```diff
@@ -8,6 +8,7 @@
 
 #import "ATUAutotypeURL.h"
 
+
```
could you remove that line?

georgesnow replied:
yup. gone

mstarke commented on AutotypeURL/ATUAutotypeURL.m:
```diff
@@ -51,14 +52,33 @@ - (nonnull NSString *)windowTitleForRunningApplication:(nonnull NSRunningApplica
   NSAppleScript *script = [[NSAppleScript alloc] initWithSource:@"tell application \"Safari\" to get URL of front document"];
-  NSAppleEventDescriptor *aed = [script executeAndReturnError:NULL];
-  return aed.stringValue;
+    NSAppleEventDescriptor *aed = [script executeAndReturnError:NULL];
```
Please apply the same code style as the project (2 spaces, no tabs)

georgesnow replied:
ugh sorry....i have habit of using tabs...fixed

mstarke commented on AutotypeURL/ATUAutotypeURL.m:
```diff
@@ -51,14 +52,33 @@ - (nonnull NSString *)windowTitleForRunningApplication:(nonnull NSRunningApplica
   return aed.stringValue;
+}
+//Additions for parsing the url subdomain.root.com
+- (NSString *)_parseURL: (NSString*) url;
```
the opening bracket should be on the same line and the trailing semicolon is a funny thing to do :)

georgesnow replied:
haha...can you tell i am not a Obj-C coder?? I spent 5 minutes trying to figure out that syntax error which resulted in this semicolon.

but now it's fixed.

mstarke commented on AutotypeURL/ATUAutotypeURL.m:
```diff
@@ -51,14 +52,33 @@ - (nonnull NSString *)windowTitleForRunningApplication:(nonnull NSRunningApplica
+//Additions for parsing the url subdomain.root.com
+- (NSString *)_parseURL: (NSString*) url;
+{
+    NSArray *separatedURL = [url componentsSeparatedByCharactersInSet:[NSCharacterSet characterSetWithCharactersInString:@"/"]];
```
Just a guess: Could you use ```NSURL``` for a more robust parsing? I would guess that the URL field always returns a valid URL so ```[[NSURL alloc] initWithString:url]``` should be useable.

georgesnow replied:
done. i was able to get this working with out too much issue.
thanks for the recommdation. I used an array cause thats how my AppleScript worked, but you are right this is far more robust.

georgesnow committed:
requested fixes

cleaned up spacing etc..changed parsing to NSURL. 
tested against Safari. Safari Title is returning correct value in Console.
however, when i test it with MacPass. MP is reporting that there is no match.
i may have improper settings and or entry setup? not sure

georgesnow commented:
@mstarke ok hopefully i think got all the requested changes fixed.

georgesnow commented:
Did a quick test against Chrome, which i didn't make any changes to. I have 3 test entries with the full URL, domain.com, sub.domain.com. MacPass is still saying no entry found. Screenshot below shows it's capturing the full URL similar to what i saw with Safari (with and without my changes).
 


![Screen Shot 2019-05-09 at 1 57 18 PM](https://user-images.githubusercontent.com/12751441/57475595-9db8b480-7262-11e9-8b0b-399318ad03ef.png)

mstarke commented:
The build of MacPass does only log the window title nothing more. So it's clear that no match is found based on the URL. I've pushed changes to the feature brach of MacPass (https://github.com/MacPass/MacPass/commit/f9b36177fd183316a6192fe2d20dca90dbdb73df) that will allow you to actually use the URL and you then should get hits.

I've also successfully created a second setup (as described in #2) to use a dedicated application to run the apple script. This allows MacPass to not require the ability to send Apple Events and only the helper applications need this capability.

georgesnow committed:
added Chrome parsing

extended the parsing to Chrome
